src/transports/winhttp.c

```diff
@@ -972,7 +972,7 @@ static int winhttp_stream_read(
 	/* Enforce a reasonable cap on the number of replays */
 	if (replay_count++ >= GIT_HTTP_REPLAY_MAX) {
 		git_error_set(GIT_ERROR_HTTP, "too many redirects or authentication replays");
-		return -1;
+		return GIT_ERROR; /* not GIT_EAUTH because the exact cause is not clear */
```

Comment:
🤔  This is an interesting one.  I wonder if we should examine all the attempts and return `GIT_EAUTH` if they were all authentication failures and `-1` otherwise.  (Similarly with the other loops.)

What do you think?

Reply:
My instinct would be to return `GIT_EAUTH` if any were authentication failures, since the other redirects are likely to be leading up to or a consequence of the auth failure. That is, it seems to me that auth failure should be sticky. Or maybe our divergent instincts mean that we should leave it as is. Either way, I'd prefer to do that as a follow-up.